shell/components/ResourceList/index.vue

```diff
@@ -119,6 +121,65 @@ export default {
 
     showIncrementalLoadingIndicator() {
       return this.perfConfig?.incrementalLoading?.enabled;
+    },
+    listLength() {
+      const { params: { resource: type } } = this.$route;
+
+      return this.$store.getters['cluster/listLength'](type);
+
+      // return undefined;
+    }
+  },
+
+  methods: {
```

Comment:
most of the "sorting, filtering, pagination" controls are actually local to sortable table which isn't especially useful when kicking off queries to the API or the worker since the actual action to do so happens up here in the resource-list. Passing down these methods provides a convenient way to get these values back up to where the action is dispatched. You're right though, these can be moved into resource-fetch but should regardless include a comment as to what they are and where they come from.